src/communication/network_interface_cl.c

```diff
@@ -1885,6 +1885,46 @@ tde_get_data_keys ()
 }
 #endif /* UNSTABLE_TDE_FOR_REPLICATION_LOG */
 
+
+int
+dblink_get_cipher_master_key ()
+{
+#if defined(CS_MODE)
+  TDE_DATA_KEY_SET data_keys;
+  int error = ER_NET_CLIENT_DATA_RECEIVE;
+  int req_error, area_size;
+  char *ptr;
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply, *area;
+
+  dblink_Cipher_key.is_loaded = false;
+
+  reply = OR_ALIGNED_BUF_START (a_reply);
+
+  req_error =
+    net_client_request2 (NET_SERVER_TDE_GET_DATA_KEYS, NULL, 0, reply,
```

Comment:
The point of what you are saying is that it would be good to have the key value independent of the TDE policy?
I understand that the <db-name>_keys file is not used by anything other than TDE.
Do you mean to write a separate routine for accessing that information?
Or is it better to get the key-related files in another way?

Reply:
_keys file is only for master keys, not the data key that you use.